packages/gateway/src/Shard.ts

```diff
@@ -129,13 +133,19 @@ export class DiscordenoShard {
     url.searchParams.set('v', this.gatewayConfig.version.toString())
     url.searchParams.set('encoding', 'json')
 
-    const socket = new WebSocket(url.toString())
+    let socket: NodeWebSocket
+
+    // @ts-expect-error Deno
+    if (isDeno) socket = new WebSocket(url.toString())
+    else if (isNode) socket = new NodeWebSocket(url.toString())
+    else throw new Error('Unsupported runtime')
```

Comment:
Should use node still. We do support other runtimes like bun which support ws. Its just deno that doesnt support ws

Reply:
Just default to node if not deno like socket = isDeno ? Websocket : Nodewebsocket

Reply:
This was what ITOH suggested to me in DM since he said bun is not fully compatible yet

Reply:
im pretty sure bun works with ws but if it doesnt we need to find a lib for it. @Tricked-dev any thoughts?